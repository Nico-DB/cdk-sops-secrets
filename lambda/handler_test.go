package main

import (
	"context"
	"encoding/json"
	"testing"
	"time"

	"github.com/aws/aws-lambda-go/cfn"
	"github.com/aws/aws-lambda-go/lambdacontext"
	"github.com/gkampitakis/go-snaps/snaps"
)

func Test_FullWorkflow_Create(t *testing.T) {
	mocks := &AWS{
		secretsmanager: &SecretsManagerMockClient{},
		s3downlaoder:   &S3ManagerMockClient{},
	}
	d := time.Now().Add(50 * time.Millisecond)
	ctx, _ := context.WithDeadline(context.Background(), d)
	ctx = lambdacontext.NewContext(ctx, &lambdacontext.LambdaContext{
		AwsRequestID:       "AwsRequestID",
		InvokedFunctionArn: "arn:aws:lambda:us-east-2:123456789012:function:cdk-sops-secrets",
	})
	inputJson := ReadJSONFromFile(t, "event_create.json")
	var event cfn.Event
	err := json.Unmarshal(inputJson, &event)

	phys, data, err := mocks.syncSopsToSecretsmanager(ctx, event)
	check(err)
	snaps.MatchSnapshot(t, phys, data, err)
}

func Test_FullWorkflow_Update(t *testing.T) {
	mocks := &AWS{
		secretsmanager: &SecretsManagerMockClient{},
		s3downlaoder:   &S3ManagerMockClient{},
	}
	d := time.Now().Add(50 * time.Millisecond)
	ctx, _ := context.WithDeadline(context.Background(), d)
	ctx = lambdacontext.NewContext(ctx, &lambdacontext.LambdaContext{
		AwsRequestID:       "AwsRequestID",
		InvokedFunctionArn: "arn:aws:lambda:us-east-2:123456789012:function:blank-go",
	})
	inputJson := ReadJSONFromFile(t, "event_update.json")
	var event cfn.Event
	err := json.Unmarshal(inputJson, &event)

	phys, data, err := mocks.syncSopsToSecretsmanager(ctx, event)
	check(err)
	snaps.MatchSnapshot(t, phys, data, err)
}

func Test_FullWorkflow_Delete(t *testing.T) {
	mocks := &AWS{
		secretsmanager: &SecretsManagerMockClient{},
		s3downlaoder:   &S3ManagerMockClient{},
	}
	d := time.Now().Add(50 * time.Millisecond)
	ctx, _ := context.WithDeadline(context.Background(), d)
	ctx = lambdacontext.NewContext(ctx, &lambdacontext.LambdaContext{
		AwsRequestID:       "AwsRequestID",
		InvokedFunctionArn: "arn:aws:lambda:us-east-2:123456789012:function:blank-go",
	})
	inputJson := ReadJSONFromFile(t, "event_delete.json")
	var event cfn.Event
	err := json.Unmarshal(inputJson, &event)

	phys, data, err := mocks.syncSopsToSecretsmanager(ctx, event)
	check(err)
	snaps.MatchSnapshot(t, phys, data, err)
}
