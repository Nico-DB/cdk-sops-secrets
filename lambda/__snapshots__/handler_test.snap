
[Test_FullWorkflow_Create - 1]
arn:custom:sopssync:eu-central-1:123456789012:secret:testsecret
map[string]interface {}{
    "ARN":           "arn:custom:sopssync:eu-central-1:123456789012:secret:testsecret",
    "Name":          "dfb70680d326c245dbdd113d635caa16113e78b1eb8b7447f10eeb861e491724",
    "VersionId":     "c81e6095f906a1b6f50f63cf51b511e59fe2061b8564473a92216ee1844d2cf6",
    "VersionStages": []*string{
        &"dfb70680d326c245dbdd113d635caa16113e78b1eb8b7447f10eeb861e491724",
    },
}
nil
---

[Test_FullWorkflow_Update - 1]
arn:custom:sopssync:eu-central-1:123456789012:secret:testsecret
map[string]interface {}{
    "ARN":           "arn:custom:sopssync:eu-central-1:123456789012:secret:testsecret",
    "Name":          "dfb70680d326c245dbdd113d635caa16113e78b1eb8b7447f10eeb861e491724",
    "VersionId":     "c81e6095f906a1b6f50f63cf51b511e59fe2061b8564473a92216ee1844d2cf6",
    "VersionStages": []*string{
        &"dfb70680d326c245dbdd113d635caa16113e78b1eb8b7447f10eeb861e491724",
    },
}
nil
---

[Test_FullWorkflow_Delete - 1]

map[string]interface {}{}
nil
---
